### ISSUE TITLE
Add confusion maps support to PauliMeasurementGate

### ISSUE THREAD

**Issue by tanujkhattar** [quantumlib/Cirq#5571]
https://github.com/quantumlib/Cirq/pull/5480 added support for confusion maps to measurement gates. The feature request is to extend this support to `cirq.PauliMeasurementGate` as well. 

P2 - we should do it in the next couple of quarters

**Comment by Tarun-Kumar07** [comment]
@tanujkhattar , can I work on this issue?
I am new to open source, in case of any queries where can I reach out for help ?

**Comment by 95-martin-orion** [comment]
@tanujkhattar was this resolved by #5480? If so, apologies for not closing it at that time.

Welcome to the community @Tarun-Kumar07!

**Comment by 95-martin-orion** [comment]
> {...} the constructors in [pauli_measurement_gates.py](https://github.com/quantumlib/Cirq/blob/master/cirq-core/cirq/ops/pauli_measurement_gate.py) don't take confusion map as parameter, but they are taken in [measurement_gate.py](https://github.com/quantumlib/Cirq/blob/master/cirq-core/cirq/ops/measurement_gate.py).

You are absolutely right - I glossed over the fact that this issue is specific to Pauli measurements. I've been away from Cirq for a while, but I suspect this is still open and useful to have. @tanujkhattar can confirm, but until then you're welcome to start working on it. Thanks!

As far as questions go, you can join the [cirq-dev googlegroup](https://groups.google.com/g/cirq-dev) to get an invite to the public Cirq sync. The meeting is a good place to ask questions, and the associated doc has links to other support options.

**Comment by Kirs67** [comment]
Hi, @Tarun-Kumar07 
Are you still working on this issue, or is it free to take?

**Comment by Tarun-Kumar07** [comment]
@Kirs67 I'm working on this issue

**Comment by tanujkhattar** [comment]
@Tarun-Kumar07 Can you please share a summary of the progress you've made so far and whether you have any questions? Also, any estimate of by when you might be able to send a PR ? This has been open for a while now and it'll be good to gain some traction here.

**Comment by Tarun-Kumar07** [comment]
Hi @tanujkhattar, Can you assign this issue to me ? 

**Progress :**
-   I tried to build a single qubit circuit and use confusion_maps in it's measurement gate
-   Got an idea on how to add confusion_maps to `PauliMeasurementGate` by looking at the changes in this [commit](https://github.com/quantumlib/Cirq/pull/5480/commits/883626ec920f9c9689bc953bfcdd86550e5120f5)

**Doubts :** 
-    As of now no, I am reading about PauliStrings.  Will mention here incase of any doubts.

**Estimate for PR :**
- I plan on submitting the PR before the next Cirq Cync i.e. 7/6/23. 

PS - I am new to quantum computing and Cirq, will take some time to get to speed 😄 .

**Comment by NoureldinYosri** [comment]
@MZuhairKhan done

**Comment by MZuhairKhan** [comment]
Hey @NoureldinYosri, I think I have the implementation mostly done (I hope). Do I also need to modify the _decompose_ function to use the confusion matrix?

**Comment by daxfohl** [comment]
Yes, defer_measurements uses this decompose function.

**Comment by Akshita07** [comment]
Hello, is someone working on this issue? If not, I want to take it up. Please assign it to me.

**Comment by Akshita07** [comment]
Thank you for the explanation @NoureldinYosri
